From a741790a6af54b46f7d35c85cf95d65a7adf8aa6 Mon Sep 17 00:00:00 2001
From: Nicolin Chen <b42378@freescale.com>
Date: Wed, 13 Nov 2013 12:05:27 +0800
Subject: [PATCH 427/505] ENGR00286961 unit-test: mxc_tuner_test: use better
 way to get pid

ps command might be different in different rootfs, it sometimes would fail to
be executed. So it's not an ideal way to get pid by using it. Thus change it.

Tested with 3.0.35 LTIB rootfs and newest Yocto rootfs.

Signed-off-by: Nicolin Chen <b42378@freescale.com>
---
 test/mxc_tuner_test/mxc_tuner_test.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/mxc_tuner_test/mxc_tuner_test.sh b/test/mxc_tuner_test/mxc_tuner_test.sh
index 6599b4b..e00fdc2 100644
--- a/test/mxc_tuner_test/mxc_tuner_test.sh
+++ b/test/mxc_tuner_test/mxc_tuner_test.sh
@@ -3,8 +3,8 @@
 num=`arecord -l |grep -i "si476" |awk '{ print $2 }'|sed 's/://'`
 
 arecord -D hw:$num,0 -f dat | aplay -D hw:0,0 -f dat &
+pid=$!
 ./mxc_tuner_test.out
-pid=$(ps | grep "arecord -D hw:$num,0 -f dat" | grep -v ' grep ' | awk '{print $1}')
 echo "killing pid = $pid"
 kill $pid
 
-- 
1.8.0

